### PR TITLE
Add reference-policy option to oc tag

### PR DESCRIPTION
**Pull request by mfojtik**

### CODE REVIEW

**Comment by mfojtik** [comment]
@miminar can I have quick review here? seems like an easy fix for great profit :-)

I tested this:

```
# Set reference policy to local
$ oc tag --source=docker openshift/origin:latest test/origin:foo --reference-policy=local

# Change reference policy to source
$ oc tag --source=docker openshift/origin:latest test/origin:foo --reference-policy=source
```

**Comment by mfojtik** [comment]
[test]

File: pkg/cmd/cli/cmd/tag.go [hunk 1]
```diff
@@ -273,6 +278,12 @@ func (o TagOptions) Validate() error {
 		return errors.New("--alias and --delete may not be both specified")
 	}
 
+	switch o.referencePolicy {
```
Comment (soltysh) [inline pkg/cmd/cli/cmd/tag.go]: Isn't plain old if better suited here, this reads weird 😉

Comment (mfojtik) [inline pkg/cmd/cli/cmd/tag.go]: done :-)

File: pkg/cmd/cli/cmd/tag.go [hunk 2]
```diff
@@ -85,12 +89,13 @@ func NewCmdTag(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Comm
...
+	cmd.Flags().StringVar(&opts.sourceKind, "source", opts.sourceKind, "Optional hint for the source type; valid values are 'imagestreamtag', 'istag', 'imagestreamimage', 'isimage', and 'docker'.")
+	cmd.Flags().BoolVarP(&opts.deleteTag, "delete", "d", opts.deleteTag, "Delete the provided spec tags.")
 	cmd.Flags().BoolVar(&opts.aliasTag, "alias", false, "Should the destination tag be updated whenever the source tag changes. Defaults to false.")
 	cmd.Flags().BoolVar(&opts.referenceTag, "reference", false, "Should the destination tag continue to pull from the source namespace. Defaults to false.")
 	cmd.Flags().BoolVar(&opts.scheduleTag, "scheduled", false, "Set a Docker image to be periodically imported from a remote repository. Defaults to false.")
 	cmd.Flags().BoolVar(&opts.insecureTag, "insecure", false, "Set to true if importing the specified Docker image requires HTTP or has a self-signed certificate. Defaults to false.")
+	cmd.Flags().StringVar(&opts.referencePolicy, "reference-policy", opts.referencePolicy, "Describes how pull-specs for images in the image stream tag are generated. Valid values are 'source' (default) or 'local'.")
```
Comment (miminar) [inline pkg/cmd/cli/cmd/tag.go]: The help doesn't help me to understand what it does.

Comment (miminar) [inline pkg/cmd/cli/cmd/tag.go]: What about `Instruct image consumers where to pull the image from. Use 'local' to pull from integrated registry. Use 'source' to pull from the image source (default).`

Comment (mfojtik) [inline pkg/cmd/cli/cmd/tag.go]: @miminar i don't think that describes what this does... basically this just says how the triggers should resolve the image... how about:

Specify how triggers should resolve the image. Use "local" to use integrated registry or "source" for original image source (default).

**Comment by mfojtik** [comment]
[merge]

File: pkg/cmd/cli/cmd/tag.go [hunk 3]
```diff
@@ -85,12 +96,13 @@ func NewCmdTag(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Comm
...
+	cmd.Flags().StringVar(&opts.sourceKind, "source", opts.sourceKind, "Optional hint for the source type; valid values are 'imagestreamtag', 'istag', 'imagestreamimage', 'isimage', and 'docker'.")
+	cmd.Flags().BoolVarP(&opts.deleteTag, "delete", "d", opts.deleteTag, "Delete the provided spec tags.")
 	cmd.Flags().BoolVar(&opts.aliasTag, "alias", false, "Should the destination tag be updated whenever the source tag changes. Defaults to false.")
 	cmd.Flags().BoolVar(&opts.referenceTag, "reference", false, "Should the destination tag continue to pull from the source namespace. Defaults to false.")
 	cmd.Flags().BoolVar(&opts.scheduleTag, "scheduled", false, "Set a Docker image to be periodically imported from a remote repository. Defaults to false.")
 	cmd.Flags().BoolVar(&opts.insecureTag, "insecure", false, "Set to true if importing the specified Docker image requires HTTP or has a self-signed certificate. Defaults to false.")
+	cmd.Flags().StringVar(&opts.referencePolicy, "reference-policy", sourceReferencePolicy, "Allow to request pull-trough for external image when set to 'local'. Defaults to 'source'.")
```
Comment (miminar) [inline pkg/cmd/cli/cmd/tag.go]: typo in `pull-trough` :smile:

**Comment by miminar** [comment]
Reference tagging original API change PR #12217

File: pkg/cmd/cli/cmd/tag.go (hunk 3 above)
Comment (mfojtik) [inline pkg/cmd/cli/cmd/tag.go]: hups

**Pull request merged**